perl-lib/OESS/lib/OESS/Cloud.pm

```diff
@@ -81,6 +88,40 @@ sub setup_endpoints {
             $peer->peer_asn($res->{AmazonSideAsn});
             push @$result, $ep;
 
+        } elsif ($ep->interface()->cloud_interconnect_type eq 'gcp-partner-interconnect') {
+            warn "oooooooooooo GCP oooooooooooo";
+            my $gcp = OESS::Cloud::GCP->new();
+
+            my $id_gen = Data::UUID->new;
+            my $id_obj = $id_gen->create();
+            my $uuid   = $id_gen->to_string($id_obj);
+
+            # GCP attachment names ($connection_id) require that: the
+            # first character must be a lowercase letter, and all
+            # following characters must be a dash, lowercase letter,
+            # or digit, except the last character, which cannot be a
+            # dash. To meet these requirements 'a-' is appended to a
+            # lowercase uuid.
+            my $interconnect_name = $vrf_name;
+            my $connection_id     = 'a-' . lc($uuid);
+
+            my $interface = $gcp->select_interconnect_interface(
+                entity => $ep->entity,
+                pairing_key => $ep->cloud_account_id
+            );
+
+            my $res = $gcp->insert_interconnect_attachment(
+                interconnect_id   => $interface->cloud_interconnect_id,
+                interconnect_name => $interconnect_name,
+                bandwidth         => 'BPS_50M',
```

Comment:
don't hard-code this... do we need to make the bandwidth selector something else for GCP interfaces?